nilearn/_utils/param_validation.py

```diff
@@ -58,30 +73,52 @@ def check_threshold(threshold, data, percentile_func, name="threshold"):
         )
         if not threshold.endswith("%"):
             raise ValueError(message)
-
         try:
-            percentile = float(threshold[:-1])
+            threshold = float(threshold[:-1])
+            percentile = True
         except ValueError as exc:
             exc.args += (message,)
             raise
+    elif not isinstance(threshold, numbers.Real):
+        raise TypeError(
+            f"{name} should be either a number "
+            "or a string finishing with a percent sign"
+        )
```

Comment:
maybe check for both str or a real number, and hoist this as part of a guard clause at the very beginning of the function

Reply:
It was actually implemented that way. I've changed it.

I first check if it is `str`, I get the number part. If it is not a number, I raise error. 

I did this to treat threshold as a number in the following parts and make it the difference in treating percentile and float more clear. 

If you prefer other way, I can change it.

Reply:
na that's ok